File: nelio-alves-algoritmo-e-logica-de-programacao/python/exercicios/alturas.py
N: int; cont: int
soma: float; media: float; porcentagem: float

N = int(input("Quantas pessoas serao digitadas? "))

nomes: [str] = [0 for x in range(N)]
idades: [int] = [0 for x in range(N)]
alturas: [float] = [0 for x in range(N)]

for i in range(N):
	print(f"Dados da {i + 1}a pessoa:")
	nomes[i] = str(input("Nome: "))
	idades[i] = int(input("Idade: "))
	alturas[i] = float(input("Altura: "))

cont = 0
soma = 0
for i in range(N):
	if idades[i] < 16:
		cont = cont + 1
		soma = soma + alturas[i]

media = soma / cont
porcentagem = (float (cont) / N) * 100.0

print(f"\nAltura media = {media:.2f}")
print(f"Pessoas com menos de 16 anos: {porcentagem:.1f}%")

for i in range(N):
	if idades[i] < 16:
		print(nomes[i])

File: nelio-alves/algoritmo-e-logica-de-programacao/java/src/exercicios/Alturas.java
package exercicios;

import java.util.Locale;
import java.util.Scanner;

public class Alturas {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, cont;
		double soma, media, porcentagem;

	    System.out.print("Quantas pessoas serao digitadas? ");
		N = sc.nextInt();

	    String[] nomes = new String[N];
	    int[] idades = new int[N];
	    double[] alturas = new double[N];

		for (int i = 0; i < N; i++) {
	    	System.out.printf("Dados da %da pessoa:\n", i + 1);
	    	System.out.print("Nome: ");
	        nomes[i] = sc.next();
	        System.out.print("Idade: ");
	        idades[i] = sc.nextInt();
	        System.out.print("Altura: ");
	        alturas[i] = sc.nextDouble();
	    }

		cont = 0;
		soma = 0;
		for (int i = 0; i < N; i++) {
			if (idades[i] < 16) {
				cont++;
			}
			soma = soma + alturas[i];
		}

		media = soma / N;
		porcentagem = ((double) cont / N) * 100.0;

	    System.out.printf("\nAltura media = %.2f\n", media);
	    System.out.printf("Pessoas com menos de 16 anos: %.1f%%\n", porcentagem);

		for (int i = 0; i < N; i++) {
			if (idades[i] < 16) {
	        	System.out.printf("%s\n", nomes[i]);
	        }
	    }

		sc.close();
	}
}
